### ISSUE TITLE
How to add an image

### ISSUE THREAD

**Issue by akuma605** [Crinsane/LaravelShoppingcart#92]
Hello, thank you for the package, I added the id, name, price and qty but how to add a string data caled 'image'?

**Comment by reppair** [comment]
Hello @akuma605. I just started learning php and laravel but I got that one working :) As the documentations says here -  https://github.com/Crinsane/LaravelShoppingcart, you can pass a $key => $value pair like so:  array('id' => '4832k', 'name' => 'Product 2', 'qty' => 1, 'price' => 10.00, 'options' => array('size' => 'large')).  @param Array        $options Array of additional options, such as 'size' or 'color'

**Comment by akuma605** [comment]
Hello, thank you for the reply, I added like that:
Cart::add(array(
			'id' => $product->id,
			'name' => $product->titre,
			'qty' => 1,
			'price' => $product->prix,
			'options' => array('image' => $product->image),
		));
but in my view I put {{$product->image}}, and nothing appear.
{{$product->id}}{{$product->name}}{{$product->qty}}{{$product->price}} work fine

**Comment by bebnev** [comment]
Because it is in options Collections you have to call options first and then the option you need.
Something like `{{ $product->options->size }}`

**Comment by reppair** [comment]
Are you are storing the path to the location of the image directly on the products_table  in some field as a 'string'? I have done this by setting a field in the product's $table in my create_products_migration like so: $table->string('image')->nullable(); Then I am able to say $product->image and it spits out the path to where my image is stored in. Try to dump() / dd() the value of this field. Or open 'php artisan tinker' and try $product =  app\Product::first(), then $product->image..

**Comment by reppair** [comment]
If you get something like "/path/to/image" string then maybe you can assign this value to a $variable and then put it in the cart's options.

**Comment by akuma605** [comment]
I store a path to my image and when I dd($product->image), it show me "jr.jpg" which is good, but the problem is to get this "jr.jpg" on my view, I put this like in the docu:
<?php echo ($product->options->has('image') ? $product->options->image :'');?>
not working

**Comment by akuma605** [comment]
{{ $product->options->size }} not working to
sorry for disturbing you guys:)

**Comment by Crinsane** [comment]
Could you please provide us with more of your code. How should we reproduce your error?

**Comment by akuma605** [comment]
can I add the key 'image' or do I have obligatory to put it on the 'options' key'?

**Comment by reppair** [comment]
well i loud say you can't rend your image be cause you have the wrong path, when you save the path in the table it should also be resolvable as well :) I had this issue not long ago. I echo it like so: <img class="feat-img" src="{{ $product->image }}" alt="Oops! Снимката липсва."> but if i dd() $product->image I get the following string  - "/images/products/2015-07-28 02:49:02-product.jpg" which is resolvable and the img gets rendered on the screen :) and here is a little trick I do: 

private function syncImg(Product $product, $request)
    {
        if ( $request->hasFile('image') )
        {
            $img = $request->file('image');
            $name = Carbon::now() . '-' . $img->getClientOriginalName();
            $img->move(public_path() . '/images/products/', $name);
            $imgPath = "/images/products/" . $name;
            $product->update(['image' => $imgPath]);
        }
That way when I call this method when adding new product or updating it I am able save the right path/to/my/image. And here is a g8 video from master Jeffrey to teach you exactly how this works :) https://laracasts.com/lessons/file-uploads-101

**Comment by Crinsane** [comment]
```php
// Route
Route::get('/', function () {
	Cart::add('1239ad0', 'Product 2', 2, 5.95, ['size' => 'large', 'image' => 'test.jpg']);

	$cart = Cart::content();

    return view('welcome', compact('cart'));
});
```

```php
// View
<table border="1">
    <thead>
        <tr>
            <th>Image</th>
            <th>Product</th>
            <th>Qty</th>
            <th>Item Price</th>
            <th>Subtotal</th>
        </tr>
    </thead>

    <tbody>

    <?php foreach($cart as $row) :?>

        <tr>
            <td>
                <p><?php echo ($row->options->has('image') ? $row->options->image : '');?></p>
            </td>
            <td>
                <p><strong><?php echo $row->name;?></strong></p>
                <p><?php echo ($row->options->has('size') ? $row->options->size : '');?></p>
            </td>
            <td><input type="text" value="<?php echo $row->qty;?>"></td>
            <td>$<?php echo $row->price;?></td>
            <td>$<?php echo $row->subtotal;?></td>
       </tr>

    <?php endforeach;?>

    </tbody>
</table>
```
*Result*
![schermafbeelding 2015-08-02 om 21 30 15](https://cloud.githubusercontent.com/assets/1297781/9027643/b2c33fb8-395d-11e5-84da-a9037ea52a65.png)

Looks fine to me...

**Comment by akuma605** [comment]
it work finally, thank you guys

**Comment by cuongmillioner** [comment]
thank you vr much, it work fine

**Comment by Chihaya1** [comment]
I am having problem with displaying  image in the cart using Crinsane/LaravelShoppingcart package.
my code for cartController is here:
![aaa](https://user-images.githubusercontent.com/44209535/86161832-966efd00-bb2d-11ea-986f-c4a90fbb6a1c.PNG)
my code for cart.index.blade.php  is here:
![bbb](https://user-images.githubusercontent.com/44209535/86161977-d0400380-bb2d-11ea-9389-44436f74f8ad.PNG)

![Control](https://user-images.githubusercontent.com/44209535/86161756-717a8a00-bb2d-11ea-8d8d-b837e5094d1e.PNG)

**Comment by webfelipemaia** [comment]
@Chihaya1 An easy way for you to solve this is to do dd ($ cartItem) in the view () and observe how the data is retrieved.
I added an item to Cart the way you did. And when I made the call that I suggested, the answer was this:

![dd](https://user-images.githubusercontent.com/2156393/86521847-3da0ac80-be2c-11ea-9c61-0b8861b75d78.png)

So if I take {{$ item-> attributes-> img}} I get the name of the image.

Another way to correct this would be for you to assign the image to the product, not to the cart item (session registration). In other words, if you have the product id, just use it to retrieve the registration at the bank and thus get the name of the image or the path, whatever.

_Note: Note: I am using the darryldecode package_
